nixos/modules/services/home-automation/zwave-js.nix

```diff
@@ -0,0 +1,158 @@
+{config, pkgs, lib, ...}:
+
+with lib;
+
+let
+  cfg = config.services.zwave-js;
+  cacheDir = "/var/cache/zwave-js";
+  runtimeDir = "/run/zwave-js";
+  defaultConfig = {
+    storage = {
+      inherit cacheDir;
+    };
+  };
+  mergedConfigFile = "${runtimeDir}/config.json";
+in {
+
+  options.services.zwave-js = {
+    enable = mkEnableOption (mdDoc "Start the zwave-js server on boot");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.zwave-js-server;
+      defaultText = literalExpression "pkgs.zwave-js-server";
+      description = mdDoc ''
+        Package for zwave-js server
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 3000;
+      description = mdDoc ''
+        Port for the server to listen on.
+      '';
+    };
+
+    serialPort = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = mdDoc ''
+        Serial port device path for Z-Wave controller.
+      '';
+      example = "/dev/ttyUSB0";
+    };
+
+    secretsConfigFile = mkOption {
+      type = types.path;
+      description = mdDoc ''
+        JSON file containing secret keys. This file will be merged with the
+        module-generated config file (taking precedence).
+
+        Z-Wave keys can be generated with:
+
+          {command}`< /dev/urandom tr -dc A-F0-9 | head -c32 ;echo`
+
+        ::: {.warning}
+        A file in the nix store should not be used since it will be readable to
+        all users.
+        :::
+      '';
+      example = ''
+        {
+          "securityKeys": {
+            "S2_Unauthenticated": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
+            "S2_Authenticated": "BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB",
+            "S2_AccessControl": "CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC",
+            "S0_Legacy": "DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD"
+          }
+        }
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.attrs;
+      default = {};
+      description = mdDoc ''
+        Extra configuration to be merged into the generated config file.
+      '';
+      example = literalExpression ''
+        {
+          storage = {
+            throttle = "fast";
+          };
+        }
+      '';
+    };
+
+    mockDriver = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Use a mock USB serial driver (mutually exclusive with serialPort option).
```

Comment:
I think this would be poor UX: if the user sets only the required options, they'd get something they almost certainly don't want (the mock driver is only useful for testing). This way they are prompted to specify what they probably want (the serial port), or to choose the mock driver explicitly.

Reply:
You can not set a default. Doesn't that fix the UX problem?

Reply:
Can you explain more? If I don't set a default for `mockDriver`, the user will have to explicitly specify it as `false` even if they don't care about the mock driver (which almost no user will care about). With the current implementation, they only need to specify it if they need it.

I guess I'm not seeing what downside of the current implementation you're trying to overcome?

Reply:
My suggestion was having no default for `serialPort` and using the mock driver if it is set to `null`. But it's your call.

Reply:
I think the unfortunate interface design choice is that `serialPort` may be `null`, which will trigger the assertion instead of the standard warning, that a required option is not set.

Ideally the mock driver wouldn't be a user-facing option in the first place, and could be configured in the test using `extraFlags`. That option is also something a regular user could benefit from.